helpers/osinfo.go

```diff
@@ -242,3 +242,43 @@ func FtraceEnabled() (bool, error) {
 	}
 	return b[0] == '1', nil
 }
+
+type LockdownMode int32
+
+func (l LockdownMode) String() string {
+	return lockdownModeToString[l]
+}
+
+const (
+	NOVALUE LockdownMode = iota
+	NONE
+	INTEGRITY
+	CONFIDENTIALITY
+)
+
+var stringToLockdownMode = map[string]LockdownMode{
+	"none":            NONE,
+	"integrity":       INTEGRITY,
+	"confidentiality": CONFIDENTIALITY,
+}
+
+var lockdownModeToString = map[LockdownMode]string{
+	NONE:            "none",
+	INTEGRITY:       "integrity",
+	CONFIDENTIALITY: "confidentiality",
+}
+
+func Lockdown() (LockdownMode, error) {
+	LockdownFile := "/sys/kernel/security/lockdown"
+	data, _ := os.ReadFile(LockdownFile)
+	dataString := string(data[:])
+
+	for lockString, lockMode := range stringToLockdownMode {
```

Comment:
Since maps aren't guaranteed to be in sequential order, is that an issue here? Can multiple values be present? (see my comment about what my file looks like)

Reply:
the file syntax is always the 3 values in the same order and the one being used is with [].

Reply:
Check https://wiki.archlinux.org/title/security#Kernel_lockdown_mode out and the lockdown manual page. There are some online discussions about it and some other projects, libe bpftrace, doing the same thing .

They go beyond and try to load small bpfs and create maps to test if other types of blockers are in place iirc.

Reply:
Ah got it, looks good then.